lib/Model.js

```diff
@@ -112,6 +127,7 @@ function Model(opts) {
 			extend_associations    : extend_associations,
 			association_properties : association_properties,
 			setupAssociations      : setupAssociations,
+                        checkAssociations      : checkAssociations,
```

Comment:
Could you please rename this to `hasAutoFetchAssociationsLoaded` ?
It's a bit long, but more indicative of what it's actually doing.